src/complete/buildin.rs

```diff
@@ -3,11 +3,46 @@ use anyhow::Result;
 use once_cell::sync::Lazy;
 use std::process::Command;
 use std::{collections::HashMap, iter::zip};
-use tower_lsp::lsp_types::{CompletionItem, CompletionItemKind, Documentation};
+use tower_lsp::lsp_types::{CompletionItem, CompletionItemKind, Documentation, InsertTextFormat};
+
+use crate::consts::TREESITTER_CMAKE_LANGUAGE;
+use crate::languageserver::get_client_capabilities;
+use crate::utils::get_node_content;
+
+/// convert input text to a snippet, if possible.
+fn convert_to_lsp_snippet(key: &str, input: &str) -> String {
+    let mut parse = tree_sitter::Parser::new();
+    parse.set_language(&TREESITTER_CMAKE_LANGUAGE).unwrap();
+    let tree = parse.parse(input, None).unwrap();
+    let mut node = tree.root_node().child(0).unwrap();
+    if node.kind_id() == 78 {
```

Comment:
I think we should move 78 to const

Reply:
done. 

BTW, do you know if it is better to avoid using these hard-coded values? 
These enumerations are defined in C source files and are not exposed in header files...

Reply:
maybe you can use kind , not kind_id